### DISCUSSION TITLE
can I use custom svgs

### DISCUSSION THREAD

**Discussion by knightswatch3** [airbnb/visx#1130] [Q&A]
I am new to the visx library. But I am curious to know more about this. 

Can I use custom svgs for the dendogram nodes ?

**Comment by williaster** [comment]
Yep! Here's a [demo](https://codesandbox.io/s/charming-goldberg-l0q4p?file=/Example.tsx).

<img src="https://user-images.githubusercontent.com/4496521/114086566-9b54a000-9867-11eb-8535-18920ee9f5f2.png" width="400" />

**Reply by knightswatch3** [reply]
@williaster Appreciate the response.

Does visx allow me to add/delete nodes on the DOM. Is that handled without the conflicts with the DOM manipulation my FE handles ? 
How smooth is it ?
Also, can I animate the nodes rendering ? deleting ?
Can you please point me to some examples or articles ?
TIA.

**Reply by williaster** [reply]
`@visx/network` is svg based, and it expects `react` to handle all of the DOM manipulation. If you're not familiar with [react](https://reactjs.org/) I'd suggest starting with that.

`@visx/network` requires you to determine the layout for your network and position your nodes + relationships yourself, it does not include any layout algorithms nor does it handle animation versus something like [`cytoscape`](http://js.cytoscape.org/) which handles it all for you. You can find all of our examples on the project website [here](https://airbnb.io/visx).

**Reply by knightswatch3** [reply]
@williaster I am experienced with React. But my question is, if I'd like to represent a DAG and have event handlings( to add,delete,modify nodes) to the dendgrams/tree/network diagrams. What do you suggest ?

Cytoscape or Visx ?

**Reply by williaster** [reply]
The hardest parts of network visualization are 
- layout calculation (dagre, etc. for a DAG view)
- performance (for large graphs where SVG is not performant and you need canvas or webgl)

Since there are other existing solutions for those (such as cytoscacpe), so far we have decided not to re-implement them in `visx`. The existing `@visx/network` package does very little, and so for your use case you would need to write a layout algorithm and potentially handle perf issues if you DAGs are large.

For a recent internal project for viewing DAGs at Airbnb I chose `react-cytoscape` which is a simple wrapper around `cytoscape`, I would probably recommend that because its so battle-tested, has great docs, and will support almost any type of visual customization you need. We integrated `visx`-based tooltips on top of it, but otherwise it worked well.
